containers/init/ready/ready.go

```diff
@@ -125,83 +118,85 @@ func findDriverPort(pod *corev1.Pod) int32 {
 	return DefaultDriverPort
 }
 
-// WaitForReadyPods blocks until pods with matching label selectors are ready.
+// WaitForReadyPods blocks until all worker pods belonging to the load test are
+// ready.
 // It accepts a context, allowing a timeout or deadline to be specified. When
 // all pods are ready, it returns a slice of strings with the IP address and
-// driver port for each matching pod. The order will match the label selectors.
-//
-// The syntax for the selectors is defined in the Parse function documented at
-// pkg.go.dev/k8s.io/apimachinery/pkg/labels.
+// driver port for each matching pod. server pod would come before client pod.
 //
 // The driver port is determined by searching the pod for a container with a TCP
 // port named "driver". If there is no port named "driver" exposed on any of the
 // matching pod's containers, the value of DefaultDriverPort will be used.
 //
 // If the timeout is exceeded or there is a problem communicating with the
 // Kubernetes API, an error is returned.
-func WaitForReadyPods(ctx context.Context, pl PodLister, sels []string) ([]string, error) {
+func WaitForReadyPods(ctx context.Context, lg LoadtestGetter, pl PodLister, testName string) ([]string, error) {
 	timeoutsEnabled := true
 	deadline, ok := ctx.Deadline()
 	if !ok {
 		timeoutsEnabled = false
 		log.Printf("no timeout is set; this could block forever")
 	}
-
-	selectors, err := parseSelectors(sels)
-	if err != nil {
-		return nil, err
-	}
-
-	var podAddresses []string
-	for range selectors {
-		podAddresses = append(podAddresses, "")
-	}
-
-	matchCount := 0
+	var loadtest *grpcv1.LoadTest
+	var clientPodAddresses []string
+	var serverPodAddresses []string
+	clientMatchCount := 0
+	serverMatchCount := 0
 	matchingPods := make(map[string]bool)
 
 	for {
 		if timeoutsEnabled && time.Now().After(deadline) {
 			return nil, errors.Errorf("deadline exceeded (%v)", deadline)
 		}
-
+		if loadtest == nil {
+			l, err := lg.Get(testName, metav1.GetOptions{})
+			if err != nil {
+				log.Printf("failed to fetch loadtest: %v", err)
+				time.Sleep(pollInterval)
+				continue
+			}
+			loadtest = l
+			for range loadtest.Spec.Clients {
+				clientPodAddresses = append(clientPodAddresses, "")
+			}
+			for range loadtest.Spec.Servers {
+				serverPodAddresses = append(serverPodAddresses, "")
+			}
+		}
 		podList, err := pl.List(metav1.ListOptions{})
 		if err != nil {
 			log.Fatalf("failed to fetch list of pods: %v", err)
 		}
-
-		for _, pod := range podList.Items {
-			if !isPodReady(&pod) {
+		ownedPods := status.PodsForLoadTest(loadtest, podList.Items)
+		for _, pod := range ownedPods {
+			if !isPodReady(pod) {
+				continue
+			}
+			if pod.GetLabels()[config.RoleLabel] == config.DriverRole {
```

Comment:
We should not need the GetLabels method. The metadata piece of the pod uses composition, so we should just be able to do `pod.Labels[config.RoleLabel]`.

Reply:
Done